cmake/StaticAnalyzers.cmake

```diff
@@ -1,100 +1,79 @@
-      message(STATUS "CPPCHECK_OPTIONS suppress: ${SUPPRESS_DIR}")
-      set(CMAKE_CXX_CPPCHECK
-          ${CPPCHECK}
-          --template=${CPPCHECK_TEMPLATE}
-          --enable=style,performance,warning,portability
-          --inline-suppr
-          # We cannot act on a bug/missing feature of cppcheck
-          --suppress=cppcheckError
-          --suppress=internalAstError
-          # if a file does not have an internalAstError, we get an unmatchedSuppression error
-          --suppress=unmatchedSuppression
-          # noisy and incorrect sometimes
-          --suppress=passedByValue
-          # ignores code that cppcheck thinks is invalid C++
-          --suppress=syntaxError
-          --suppress=preprocessorErrorDirective
-          --inconclusive
-          --suppress=${SUPPRESS_DIR})
-    else()
-      # if the user provides a CPPCHECK_OPTIONS with a template specified, it will override this template
-      set(CMAKE_CXX_CPPCHECK ${CPPCHECK} --template=${CPPCHECK_TEMPLATE} ${CPPCHECK_OPTIONS})
+      set(SUPPRESS_DIR "${CMAKE_CURRENT_BINARY_DIR}/_deps/*.h")
+      message(STATUS "CPPCHECK suppressing warnings for directory: ${SUPPRESS_DIR}")
+      set(CPPCHECK_OPTIONS
+        --enable=style,performance,warning,portability
+        --template=${CPPCHECK_TEMPLATE}
+        --inline-suppr
+        --inconclusive
+        --suppress=cppcheckError  # Suppress cppcheck errors
+        --suppress=internalAstError # Suppress internal AST errors
+        --suppress=unmatchedSuppression # Suppress unmatched suppressions in the code
+        --suppress=passedByValue # Suppress passed by value warnings
+        --suppress=syntaxError # Suppress syntax errors in the code
+        --suppress=preprocessorErrorDirective # Suppress preprocessor error directives
+        --suppress=${SUPPRESS_DIR})
     endif()
-    if(NOT
-       "${CMAKE_CXX_STANDARD}"
-       STREQUAL
-       "")
-      set(CMAKE_CXX_CPPCHECK ${CMAKE_CXX_CPPCHECK} --std=c++${CMAKE_CXX_STANDARD})
+
+    if(NOT "${CMAKE_CXX_STANDARD}" STREQUAL "")
+      list(APPEND CPPCHECK_OPTIONS --std=c++${CMAKE_CXX_STANDARD})
     endif()
+
     if(${WARNINGS_AS_ERRORS})
-      list(APPEND CMAKE_CXX_CPPCHECK --error-exitcode=2)
+      list(APPEND CPPCHECK_OPTIONS --error-exitcode=2)
     endif()
+
+    set(CMAKE_CXX_CPPCHECK ${CPPCHECK} ${CPPCHECK_OPTIONS})
   else()
-    message(${WARNING_MESSAGE} "cppcheck requested but executable not found")
+    message(WARNING "cppcheck requested but executable not found")
   endif()
 endmacro()
 
 macro(openturbine_enable_clang_tidy target WARNINGS_AS_ERRORS)
-
   find_program(CLANGTIDY clang-tidy)
   if(CLANGTIDY)
-    if(NOT
-       CMAKE_CXX_COMPILER_ID
-       MATCHES
-       ".*Clang")
-
+    if(NOT CMAKE_CXX_COMPILER_ID MATCHES ".*Clang")
       get_target_property(TARGET_PCH ${target} INTERFACE_PRECOMPILE_HEADERS)
-
-      if("${TARGET_PCH}" STREQUAL "TARGET_PCH-NOTFOUND")
+      if(NOT TARGET_PCH)
```

Comment:
Did you test that this change works like the previous version?  I based things like this off of a template that's been tested on a bunch of platforms, since we don't have an easy way to test these type of edge cases ourselves.

Reply:
I did not test it, the checks looked a little roundabout way of doing things. Anyways, reverted this change.